src/components/PDFView/index.js

```diff
@@ -146,33 +168,25 @@ class PDFView extends Component {
      * @param {Number} pageIndex
      * @returns {Number}
      */
-    calculatePageHeight(pageIndex) {
-        if (this.state.pageViewports.length === 0 || _.some(this.state.pageViewports, (viewport) => !viewport)) {
-            Log.warn('Dev error: calculatePageHeight() in PDFView called too early');
-
-            return 0;
-        }
-
-        const pageViewport = this.state.pageViewports[pageIndex];
-        const pageWidth = this.calculatePageWidth();
-        const scale = pageWidth / pageViewport.width;
-        const actualHeight = pageViewport.height * scale + PDFViewConstants.PAGE_BORDER * 2;
+    const calculatePageHeight = useCallback(
+        (pageIndex) => {
+            const viewports = pageViewports.current;
+            if (viewports.length === 0 || _.some(viewports, (viewport) => !viewport)) {
+                Log.warn('Dev error: calculatePageHeight() in PDFView called too early');
 
-        return actualHeight;
-    }
+                return 0;
+            }
 
-    /**
-     * Calculates a proper page width.
-     * It depends on a screen size. Also, the app should take into account the page borders.
-     * @returns {Number}
-     */
-    calculatePageWidth() {
-        const pdfContainerWidth = this.state.containerWidth;
-        const pageWidthOnLargeScreen = Math.min(pdfContainerWidth - PDFViewConstants.LARGE_SCREEN_SIDE_SPACING * 2, variables.pdfPageMaxWidth);
-        const pageWidth = this.props.isSmallScreenWidth ? this.state.containerWidth : pageWidthOnLargeScreen;
+            const pageViewport = viewports[pageIndex];
+            const pageWidth = calculatePageWidth();
+            const scale = pageWidth / pageViewport.width;
+            const actualHeight = pageViewport.height * scale + PDFViewConstants.PAGE_BORDER * 2;
 
-        return pageWidth + PDFViewConstants.PAGE_BORDER * 2;
-    }
+            return actualHeight;
+        },
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [numPages, calculatePageWidth],
```

Comment:
Actually we should have `pageViewports` as its dependency.
But `pageViewports` is an object(array) and changed every time the document is loaded. Having it as a state and specifying as the dependency of this method would re-render the component every time the document is loaded and would cause the component infinite re-render. And having this as a ref wouldn't change the `calculatePageWidth` method after the document is loaded.

As you can see, `pageViewports` and `numPages` are set when the document is loaded, but `numPages` is a scalar value and not changed as long as the document is not changed. So specifying this as a dependency instead of `pageViewports` would prevent the infinite re-render. And when the document is changed, it would change as well

This is the reason I added `numPages` as a dependency instead of `pageViewports`

Reply:
done